crypto/rand/drbg_ctr.c

```diff
@@ -366,8 +366,10 @@ __owur static int drbg_ctr_generate(RAND_DRBG *drbg,
         ctr32 = GETU32(ctr->V + 12) + blocks;
         if (ctr32 < blocks) {
             /* 32-bit counter overflow into V. */
-            blocks -= ctr32;
-            buflen = blocks * 16;
+            if (ctr32 != 0) {
+                blocks -= ctr32;
+                buflen = blocks * 16;
+            }
             ctr32 = 0;
```

Comment:
Should the ctr32 = 0 be moved inside the if?

Reply:
yes. good point.